### PR TITLE
redirecting login to https

### PR DESCRIPTION
**Pull request by akp2603**
### What does this code do, and why?
Redirects to https in case the original url contains http, in case of login.
Issue: #461
### How is this code tested?
Manually.

### Are any database migrations required by this change?
None

### Are there any configuration or environment changes needed?
None

### CODE REVIEW

File: app/controllers/sessions_controller.rb [hunk 1]
```diff
@@ -1,4 +1,9 @@
 class SessionsController < ApplicationController
+
+  before_action :redirect_https, only: [
+    :login
```
Comment (anaulin) [inline app/controllers/sessions_controller.rb:4]: I think it makes sense to always redirect to https, not just for login. It's the more privacy-respecting option. Let's remove the `only` clause from here, @akp2603

**Review by anaulin — CHANGES_REQUESTED** [review]

**Comment by anaulin** [comment]
A spec to verify this behavior would be nice, but not required. 💚

File: app/controllers/sessions_controller.rb (hunk 1 above)
Comment (akp2603) [reply app/controllers/sessions_controller.rb:4]: That would be a much bigger change. I am a little new to this adding ssl redirection. IMO, let's just do it for this action this time and if it works perfectly, we can add the before action in the application controller only. (That's actually the reason I kept in application_controller. Even I think this should be on all the actions. But let's just try it out with this one first. Thoughts?)

**Comment by akp2603** [comment]
I am making some changes to the failing specs. I think that should cover it.

**Comment by akp2603** [comment]
Hi @anaulin , the redirection works but a few specs were failing. Some of which I fixed. One I am stuck at. Can you please look into it and let me know what am I missing?

File: app/controllers/sessions_controller.rb (hunk 1 above)
Comment (anaulin) [reply app/controllers/sessions_controller.rb:4]: That's fair. I'm fine leaving it small for now.

**Comment by anaulin** [comment]
For the record, these are the errors that I see in the specs:

Screen Shot 2020-10-18 at 11 17 11 AM

It looks like Capybara isn't good at dealing with HTTPS in a local environment.  Apparently it is possible to set up Capybara to work perfectly with HTTPS (e.g. according to this article: https://medium.com/doctolib/https-integration-tests-with-rails-and-capybara-61fea7425e49), but for now perhaps the easiest thing to do would be to turn off the HTTPS redirect in tests. What do you think, @akp2603 , maybe add an "unless env = test" or something? (Feels a little dirty, but I think a "proper" fix will likely be a bunch more work. We can file a separate issue to get this working in specs.)

**Comment by akp2603** [comment]
> We can file a separate issue to get this working in specs.

Sounds perfect.

**Review by anaulin — APPROVED** [review]

**Pull request merged**